core/src/main/scala/kafka/log/AbstractIndex.scala

```diff
@@ -108,32 +107,42 @@ abstract class AbstractIndex(@volatile private var _file: File, val baseOffset:
   @volatile
   protected var mmap: MappedByteBuffer = {
     val newlyCreated = file.createNewFile()
-    val raf = if (writable) new RandomAccessFile(file, "rw") else new RandomAccessFile(file, "r")
+
+    val options = if (writable) List(StandardOpenOption.READ, StandardOpenOption.WRITE, StandardOpenOption.SPARSE) else List(StandardOpenOption.READ, StandardOpenOption.SPARSE)
```

Comment:
You don't need StandardOpenOption.SPARSE for read-only mode since that option is only applicable as a hint to operating system when creating new files. see: https://docs.oracle.com/javase/7/docs/api/java/nio/file/StandardOpenOption.html#SPARSE